### PR TITLE
Merge benchmarks back into main repo

### LINKED ISSUE #7625: Bring back Benchmarks
We recently decided to bring the Benchmarks tests back into this repo.

### PR DESCRIPTION
**Pull request by rowanmiller**
Resolves #7625

In a non-perf run (i.e. when you just run locally, and on CI) the tests take 68sec to execute on my laptop - so probably under a minute on most hardware.

@bricelam could you take a quick look over this one. It is mostly just a copy of the code except I changed assembly/project names (and namespaces) around.
* Benchmarks.Core (shared infra) => Microsoft.EntityFrameworkCore.Benchmarks
* Benchmarks (EF Core benchmarks) => Microsoft.EntityFrameworkCore.Benchmarks.EFCore
* Benchmarks.EF6 (EF6 benchmarks) => Microsoft.EntityFrameworkCore.Benchmarks.EF6

I'll also add a comment to the other piece of code I changed.

### CODE REVIEW

File: test/Microsoft.EntityFrameworkCore.Benchmarks/BenchmarkTestCaseRunner.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            return config["computerName"];
+        }
+
+        private static string GetFramework()
+        {
+#if NET452
+            return ".NET Framework";
+#else
+            return ".NET Core";
+#endif
```
Comment (rowanmiller) [inline test/Microsoft.EntityFrameworkCore.Benchmarks/BenchmarkTestCaseRunner.cs]: This used to use `Microsoft.Extensions.Internal.RuntimeEnvironment.RuntimeType` but that now returns `CoreCLR` even when running on the `net452` output in the xunit console runner. What I have now is what we used to have before we split the benchmarks out to a separate repo (where the compilation of tests may not have matched the compilation of product).

File: test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Query/NavigationsQueryTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        private readonly AdventureWorksFixture _fixture;
+
+        public NavigationsQueryTests(AdventureWorksFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Benchmark]
+        [AdventureWorksDatabaseRequired]
```
Comment (smitpatel) [inline test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Query/NavigationsQueryTests.cs]: Do we need attribute on each method if the class is already marked with it?

Comment (rowanmiller) [inline test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Query/NavigationsQueryTests.cs]: Possibly not, I'll play around with that after I've got the code merged. I wanted to move it over with minimal changes to start with 😄

**Comment by ErikEJ** [comment]
This is great, looking forward to try them out with the SQLCE provider

File: test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Microsoft.EntityFrameworkCore.Benchmarks.EF6.csproj [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="..\..\tools\EntityFramework.props" />
+  <PropertyGroup>
+    <TargetFrameworks>net452</TargetFrameworks>
```
Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Microsoft.EntityFrameworkCore.Benchmarks.EF6.csproj]: Might be able to get away with just `<TargetFramework>` (no *s*) here if the EF and ASP.NET targets can handle that.

Comment (bricelam) [inline test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Microsoft.EntityFrameworkCore.Benchmarks.EF6.csproj]: Might also need to exclude this from non-Windows. @natemcmaster probably knows how to do that...

Comment (natemcmaster) [inline test/Microsoft.EntityFrameworkCore.Benchmarks.EF6/Microsoft.EntityFrameworkCore.Benchmarks.EF6.csproj]: Example: https://github.com/aspnet/EntityFramework.Tools/blob/dev/build/repo.props

**Comment by bricelam** [comment]
LGTM (Just looked at `*.csproj` files)

**Pull request merged**